src/components/Clock.tsx

```diff
@@ -0,0 +1,49 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string;
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends Component<Props, State> {
+  state: Readonly<State> = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+
+      // eslint-disable-next-line
+      console.log(this.state.date.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
```

Comment:
i think you need to pass the `renamingTimerId` to `Clock`, so it won't change its name when removed 🤔

Reply:
> i think you need to pass the `renamingTimerId` to `Clock`, so it won't change its name when removed 🤔

Please, clarify, what do you mean? I have two different timers😅One in the Clock component that update time every second(timerId variable to store setInterval and clear it when we have no clock). Second in App to update clock name every 3300ms(the same name for variable)